src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -300,6 +302,15 @@ private boolean runRepair() {
 
       LazyInitializer<Set<String>> busyHosts = new BusyHostsInitializer(cluster);
 
+      // If we're using a distributed storage, we need to synchronize with other Reaper instances
+      // So we don't start too many segments at the same time
+      if (!lockSegmentRunners()) {
```

Comment:
this needs to be moved to happen after line 334 ( https://github.com/thelastpickle/cassandra-reaper/pull/622/files#diff-e6aa6503be815f976c3ed7a845621f4aR334 )
with a check first against `STARTED`

and the log line raised to `WARN` (so we get the feedback over time how required this locking is)

Reply:
i will provide a PR for this.

Reply:
https://github.com/thelastpickle/cassandra-reaper/pull/669